runtime/js/41_prompt.js

```diff
@@ -1,79 +1,95 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 const core = globalThis.Deno.core;
-const primordials = globalThis.__bootstrap.primordials;
+const ops = core.ops;
 import { isatty } from "ext:runtime/40_tty.js";
 import { stdin } from "ext:deno_io/12_io.js";
-const { ArrayPrototypePush, StringPrototypeCharCodeAt, Uint8Array } =
-  primordials;
-const LF = StringPrototypeCharCodeAt("\n", 0);
-const CR = StringPrototypeCharCodeAt("\r", 0);
-
-function alert(message = "Alert") {
-  if (!isatty(stdin.rid)) {
-    return;
-  }
+const primordials = globalThis.__bootstrap.primordials;
+const { Uint8Array, StringFromCodePoint } = primordials;
 
-  core.print(`${message} [Enter] `, false);
+const ESC = "\x1b";
+const CTRL_C = "\x03";
+const CTRL_D = "\x04";
 
-  readLineFromStdinSync();
+const bold = ansi(1, 22);
+const italic = ansi(3, 23);
+const yellow = ansi(33, 0);
+function ansi(start, end) {
+  return (str) => ops.op_use_color() ? `\x1b[${start}m${str}\x1b[${end}m` : str;
 }
 
-function confirm(message = "Confirm") {
+function alert(message = "Alert") {
   if (!isatty(stdin.rid)) {
-    return false;
+    return;
   }
 
-  core.print(`${message} [y/N] `, false);
+  core.print(
+    `${yellow(bold(`${message}`))} [${italic("Press any key to continue")}] `,
+  );
 
-  const answer = readLineFromStdinSync();
+  try {
+    stdin.setRaw(true);
+    stdin.readSync(new Uint8Array(1024));
```

Comment:
If we're supposed to read only a single key here, why do we need a buffer with size 1024? Could you also explain why you're setting the terminal to raw mode here?

Reply:
> If we're supposed to read only a single key here, why do we need a buffer with size 1024? 

A single logical keypress (most notably, combos such as shift+insert/ctrl+v) can send an arbitrarily large amount of input to stdin. 1024 is just an arbitrary size that should typically be large enough to avoid problems but small enough to avoid undue memory usage.

For example, if we used a smaller buffer size like `10`:

```js
const { stdin } = Deno

const b = new Uint8Array(10)
stdin.setRaw(true)
stdin.readSync(b)
```

That can be broken (Deno exits with `Error: Io(Kind(InvalidData))`) by copying `字字字字` to the clipboard and pasting it with the next keypress, because the 3 UTF-8 bytes of the last `字` lie across the boundary.

> Could you also explain why you're setting the terminal to raw mode here?

Raw mode is necessary to capture ESC keypresses and to progress immediately after a key is pressed.